lib/hammer_cli_foreman/commands.rb

```diff
@@ -30,7 +30,9 @@ def self.resource_config
     config[:headers] = { "Accept-Language" => HammerCLI::I18n.locale }
     config[:language] = HammerCLI::I18n.locale
     config[:timeout] = HammerCLI::Settings.get(:foreman, :request_timeout)
-    config[:timeout] = -1 if (config[:timeout] && config[:timeout].to_i < 0)
+    if (config[:timeout] && config[:timeout].to_i < 0)
+      config[:timeout] = (RestClient.version < '2.0.0') ? -1 : nil
```

Comment:
I think this logic would be best in apipie-bindings, would you mind moving it there? apipie-bindings has more version-dependent code and it'd also allow you to fix the deprecation warning to use `:read_timeout` instead of `:timeout`.

apipie-bindings could accept either nil or negative numbers and handle them correctly, abstracting all of these changes away from Hammer.